qcelemental/tests/test_models_results.py

```diff
@@ -163,3 +194,21 @@ def test_wavefunction_return_result_pointer(wavefunction_data_fixture):
         qcel.models.Result(**wavefunction_data_fixture)
 
     assert "does not exist" in str(e.value)
+
+
+@pytest.mark.parametrize("keep, indices", [
+    (None, [0, 1, 2, 3, 4]),
+    ('all', [0, 1, 2, 3, 4]),
+    ('initial_and_final', [0, 4]),
+    ('final', [4]),
+    ('none', []),
+])
+def test_optimization_trajectory_protocol(keep, indices, optimization_data_fixture):
+
+    if keep is not None:
+        optimization_data_fixture["protocols"] = {"keep_trajectory": keep}
+    opt = qcel.models.Optimization(**optimization_data_fixture)
+
+    assert len(opt.trajectory) == len(indices)
+    for result, index in zip(opt.trajectory, indices):
+        assert result.return_result == index
```

Comment:
isn't lhs an energy and rhs an integer?

Reply:
We explicitly set lhs to be an index integer for this test.